packages/vite/src/utils/generator-utils.ts

```diff
@@ -89,119 +69,59 @@ export function findExistingTargetsInProject(
     '@nrwl/js:tsc',
   ];
 
-  const arrayOfNxViteExecutors = [
-    '@nrwl/vite:build',
-    '@nrwl/vite:dev-server',
-    '@nrwl/vite:test',
-  ];
-
   // First, we check if the user has provided a target
   // If they have, we check if the executor the target is using is supported
   // If it's not supported, then we set the unsupported flag to true for that target
 
-  if (userProvidedTargets?.build) {
-    if (
-      arrayOfSupportedBuilders.includes(
-        targets[userProvidedTargets.build]?.executor
-      )
-    ) {
-      validFoundBuildTarget = userProvidedTargets.build;
-    } else {
-      unsupportedUserProvidedTargetBuild = true;
+  function checkUserProvidedTarget(target: string, executors: string[]) {
+    if (userProvidedTargets?.[target]) {
+      if (executors.includes(targets[userProvidedTargets[target]]?.executor)) {
+        output.validFoundTargetName[target] = userProvidedTargets[target];
+      } else {
+        output.userProvidedTargetIsUnsupported[target] = true;
+      }
     }
   }
 
-  if (userProvidedTargets?.serve) {
-    if (
-      arrayOfSupportedServers.includes(
-        targets[userProvidedTargets.serve]?.executor
-      )
-    ) {
-      validFoundServeTarget = userProvidedTargets.serve;
-    } else {
-      unsupportedUserProvidedTargetServe = true;
-    }
-  }
+  checkUserProvidedTarget('build', supportedBuilders);
+  checkUserProvidedTarget('serve', supportedServers);
+  checkUserProvidedTarget('test', supportedTesters);
 
-  if (userProvidedTargets?.test) {
-    if (
-      arrayOfSupportedServers.includes(
```

Comment:
`arrayOfSupportedServers` should have been `arrayOfSupportedTesters` here.
Fixed by the PR.